include/range/v3/detail/prologue.hpp

```diff
@@ -45,3 +45,6 @@ RANGES_DIAGNOSTIC_KEYWORD_MACRO
 #define declval(...) static_cast<__VA_ARGS__(*)()>(nullptr)()
 #endif
 /// \cond
+
+#pragma push_macro("I")
```

Comment:
Pragmas are all implemented defined. Do we know any compilers that _don't_ support `push/pop_macro`?

Reply:
I tried this out on all clang, gcc, and msvc and it all seemed to work. That definitely seems like a better bet than... renaming all the `I`s. 

If this doesn't work, then uh...

Reply:
lol it looks like the Windows build is failing because just including `<complex.h>` is deprecated? Am I understanding that correctly?

I have no idea why some clang builds are failing and some not.

Reply:
> I have no idea why some clang builds are failing and some not.

libc++ builds are failing, libstdc++ builds are not.

Reply:
> lol it looks like the Windows build is failing because just including `<complex.h>` is deprecated? Am I understanding that correctly?

C++17 said that "`#include <complex.h>` is equivalent to `#include <ccomplex>`", and that `<ccomplex>` "behaves as if it simply includes the header `<complex>`". C++20 said that "`#include <complex.h>` is equivalent to `#include <complex>`", and deprecated `<ccomplex>`. We've implemented the C++17 redirect, and the deprecation from C++20, but apparently not yet the C++20 redirect. (The UCRT never services old releases for anything short of catastrophe, so the redirect will eventually appear in some new version of Windows.)

TLDR: `#include <complex.h>` on Windows includes `<ccomplex>` which tells you that it's deprecated and then includes `<complex>`. In another ten years or so, `#include <complex.h>` on Windows will simply include `<complex>` directly.

Reply:
TLDR: The "bug1729" test _should_ fail and return `1` on C++17-or-later.

Reply:
The fact that it's failing on libc++ in C++14 mode suggests that libc++ implemented the C++17 change unconditionally (as did MSVC). The fact that it passes on libstdc++ suggests that libstdc++ has not implemented the C++17 change.

Reply:
Maybe @jwakely can comment?

Reply:
Well as long as it literally compiles (which it did not before) is a success, so maybe I'll just change the test to `return 0;` and add defines for MSVC to not give me the depreciation warnings?

Reply:
`#define _SILENCE_CXX17_C_HEADER_DEPRECATION_WARNING` before including anything will silence the MSVC deprecation warning.

Reply:
Looks like everything works now 🎉